repo/git.go

```diff
@@ -232,7 +240,115 @@ func Commit(localPath string, idx *git.Index) error {
 
 }
 
-// Push pushes all local commits to theh default remote & branch
+// Pull pulls all remote commits from the default remote & branch
+// (git pull)
+func Pull() error {
+	repository, err := git.OpenRepository(".")
+	if err != nil {
+		return err
+	}
+
+	origin, err := repository.Remotes.Lookup("origin")
+	if err != nil {
+		return err
+	}
+
+	// Fetch
+	cbs := &git.RemoteCallbacks{
+		CredentialsCallback:      makeCredsCB(),
+		CertificateCheckCallback: certCB,
+	}
+	fetchopts := &git.FetchOptions{RemoteCallbacks: *cbs}
+
+	err = origin.Fetch([]string{}, fetchopts, "")
+	if err != nil {
+		return err
+	}
+
```

Comment:
I think git stores the to-be-merged-from-head in `FETCH_HEAD` for the duration of the merge. Could be nice to do that in the future too.